src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -137,7 +144,11 @@ private void SelectBlobTransactions(IEnumerable<Transaction> blobTransactions, B
                 checkedBlobTransactions++;
 
                 ulong txBlobGas = (ulong)(blobTx.BlobVersionedHashes?.Length ?? 0) * _eip4844Config.GasPerBlob;
-                if (txBlobGas > _eip4844Config.MaxBlobGasPerBlock - blobGasCounter)
+
+                UInt256? blobGasLeft = spec.IsEip7742Enabled
+                    ? payloadAttributes?.MaxBlobCount * Eip4844Constants.GasPerBlob - blobGasCounter
+                    : _eip4844Config.MaxBlobGasPerBlock - blobGasCounter;
```

Comment:
Now I see this can be simplified to:
```suggestion
                UInt256 blobGasLeft = maxBlobGas - blobGasCounter;
```